### PR TITLE
Allow unix_socket_path in RedisSettings

### PR DESCRIPTION
**Pull request by lazToum**

### CODE REVIEW

File: tests/requirements-testing.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 coverage==6.3
 dirty-equals==0.1
+redislite==6.2.805324
```
Comment (samuelcolvin) [inline tests/requirements-testing.txt:3]: I don't think we need this, if we do, it's a different discussion and PR.

**Review by samuelcolvin — COMMENTED** [review]
otherwise looking good.

File: setup.py [hunk 2]
```diff
@@ -55,8 +55,8 @@
         arq=arq.cli:cli
     """,
     install_requires=[
-        'redis>=4.2.0rc2',
-        'click>=6.7',
+        'redis>=4.2.0',
+        'click>=6.7,<8.1.0',
```
Comment (samuelcolvin) [inline setup.py:59]: is this required?

File: tests/requirements-testing.txt (hunk 1 above)
Comment (lazToum) [reply tests/requirements-testing.txt:3]: This is only for testing. I haven't found any easy way to include on the CI workflow a redis-server that is listening on a unix socket.

File: setup.py (hunk 2 above)
Comment (lazToum) [reply setup.py]: The `click>=6.7,<8.1.0`, no, not anymore. There was an issue with black (https://github.com/psf/black/issues/2964). But it looks it's ok now. I removed it. The `redis>=4.2.0` is needed if we include redislite on tests.

File: tests/requirements-testing.txt (hunk 1 above)
Comment (samuelcolvin) [reply tests/requirements-testing.txt:3]: Ok, that's fine then.

**Comment by samuelcolvin** [comment]
looks like you have conflicts, hence tests aren't running.

**Pull request merged**